### PR TITLE
NIFI-3543 Fixed a bug that nifi-jms-cf-service cannot have a variable as the brokerURL.

### PR DESCRIPTION
**Pull request by ShellyLC**

### CODE REVIEW

**Comment by mosermw** [comment]
@ShellyLC If you wouldn't mind opening a NIFI JIRA ticket for this bug, we would really appreciate it!

File: nifi-nar-bundles/nifi-jms-bundle/nifi-jms-cf-service/src/main/java/org/apache/nifi/jms/cf/JMSConnectionFactoryProvider.java [hunk 1]
```diff
@@ -178,7 +178,7 @@ private void setConnectionFactoryProperties(ConfigurationContext context) {
             } else {
                 if (propertyName.equals(BROKER)) {
                     if (context.getProperty(CONNECTION_FACTORY_IMPL).evaluateAttributeExpressions().getValue().startsWith("org.apache.activemq")) {
-                        this.setProperty("brokerURL", entry.getValue());
+                        this.setProperty("brokerURL", context.getProperty(descriptor).evaluateAttributeExpressions().getValue());
```
Comment (olegz) [inline nifi-nar-bundles/nifi-jms-bundle/nifi-jms-cf-service/src/main/java/org/apache/nifi/jms/cf/JMSConnectionFactoryProvider.java]: Unfortunately this only addresses the IF and not the ELSE where the value is treated differently to account for providers other then ActiveMQ. So, if we're adding EL support for  'brokerURL' we should do it for all. 

Also, with EL one must account for a possibility of null.

Anyway, if please address it when you get a chance or let us know if you don't have time and one of us can essentially collaborate and address the comments above.

**Comment by olegz** [comment]
@ShellyLC also, please amend your commit message preceding it with the JIRA number so it could be automatically linked in JIRA (i.e., "NIFI-3543 Fixed a bug that nifi. . ."). Also, please read my comments on the differences between BUG and IMPROVEMENT.

File: nifi-nar-bundles/nifi-jms-bundle/nifi-jms-cf-service/src/main/java/org/apache/nifi/jms/cf/JMSConnectionFactoryProvider.java (hunk 1 above)
Comment (ShellyLC) [inline nifi-nar-bundles/nifi-jms-bundle/nifi-jms-cf-service/src/main/java/org/apache/nifi/jms/cf/JMSConnectionFactoryProvider.java]: Thank you for your comments. I'll address the ELSE and account for the possibility of null.

**Comment by olegz** [comment]
Thanks you @ShellyLC , will review and merge.

**Pull request closed without merging**